scipy/special/cephes/hyp2f1.c

```diff
@@ -590,3 +595,27 @@ static double hyp2f1ra(double a, double b, double c, double x,
 
     return f0;
 }
+
+
+/*
+    15.4.2 Abramowitz & Stegun.
+*/
+static double hyp2f1_neg_c_equal_bc(double a, double b, double x)
+{
+    double k;
+    double err;
+    double collector = 1;
+    double sum = 1;
+
+    for (k = 1; k <= -b; k++) {
+  	collector *= (a + k - 1)*x/k;
+    	sum += collector;
+    }
+
+    err = 1e-16 * (1 + fabs(collector)/fabs(sum));
```

Comment:
I think you need to do `if (fabs(collector) > max_collector) max_collector = collector;` inside the loop, unless you can prove the largest term is found at the end (not obvious).

Reply:
I thought about this for a minute to make sure that I was (hopefully correct). But here goes:

1. Ignore "x" because it is constant throughout the loop.
2. Bracket "a" from -1 to 1 from the above program conditions
3. Simplify the expression: the numerator is from "1 + k - 1" = "k" to "-1 + k - 1" = "k - 2"
4. See that in the first case, it's "k/k"= constant (first term = last term)
5. See that in the second case, it's "(k - 2)/k", which has a limit behavior of 1 (starts smaller and grows).

In summary, it seems to start out small and then approach 1 in all cases, just from different starting places.

Reply:
Also, I don't think that the series is alternating.

Reply:
For `x < 0` the series is alternating.

Step 1. above sounds problematic. Better to move the check inside the loop, so we don't need to reason about this.

Reply:
Even if it is alternating, it would still tend to approach 1/-1.

Reply:
It's ultimately a (k - 2)/k, (k - 1.5)/k, (k - 1)/k, k/k problem. And we're comparing the absolute value, so the negative sign can be dropped.